### ISSUE TITLE
parseUser.currentUser() & .getCurrentUserFromServer()

### ISSUE THREAD

**Issue by ScanEat** [phillwiggins/flutter_parse_sdk#24]
Hello,

I copy/paste the project example, put my applications constants.

I sign up a user and login it. (I have the user in database with the session, it's good)

I restart my application and do my own code :
`checkCurrentUser() async {
    // Best practice for starting the app. This will check for a
    var user = ParseUser.currentUser();
    user = await user.getCurrentUserFromServer();
    print("user: $user");
}`

But in my terminal, I have this :
`I/flutter (13889):  
I/flutter (13889): ----
I/flutter (13889): Hello Flutter API Response (_User : ParseApiRQ.currentUser) :
I/flutter (13889): Status Code: 200
I/flutter (13889): Payload: {objectId: LjTOWTPE8D, createdAt: 2019-01-10T12:51:19.140Z, email: TestFlutterSDK@gmail.com, username: TestFlutter, updatedAt: 2019-01-10T12:51:19.140Z, ACL: {*: {read: true}, LjTOWTPE8D: {read: true, write: true}}, sessionToken: r:5f7045a813f163282ac8a765eabf8cb6, __type: Object, className: _User}
I/flutter (13889): ----
I/flutter (13889): user: Username: null 
I/flutter (13889): Email Address:null`

The problem is that it seems that the user is retrieved but when I print my user variable, it's null.

In advance, thank you.

**Comment by phillwiggins** [comment]
Should be fixed now.

**Comment by ZeroNilZero** [comment]
why ParseUser.currentUser() return Future? 
it make a lot headache!

**Comment by maaeps** [comment]
@ZeroNilZero 
Almost everything in this SDK returns a future. As futures are a great way of dealing with asynchronous actions and even provide a way for handling errors.

If you have a case where using a future is not possible, you might have a lock at [this](https://api.dart.dev/stable/2.0.0/dart-cli/waitFor.html).
But I would not recommend using it in production, as such a case should be the result of a design flaw.

**Comment by ZeroNilZero** [comment]
@maaeps 
How can we access and use currentUser in custom ParseObject? and do some action base on it?

**Comment by maaeps** [comment]
I assume you have a custom ParseUser class as described [here](https://github.com/parse-community/Parse-SDK-Flutter#custom-objects) (but with extending ParseUser) class and want to get the current user as an instance of your custom class.

You can use provide a `parseUserConstructor` (only available in [1.0.27](https://github.com/parse-community/Parse-SDK-Flutter/tree/release/1.0.27#custom-objects)) or you can use `ParseUser.currentUser(customUserObject: YourCustomUserClass())`

**Comment by ZeroNilZero** [comment]
@maaeps 
I'm not talking about custom USER class, I want to use current user in another Object Class and do something like this:
`ParseObject get O => this.user.objectId == ParseUser.currentUser().objectId ? A : B;`
which A and B are Pointer ParseObject.

**Comment by ZeroNilZero** [comment]
> I assume you have a custom ParseUser class as described [here](https://github.com/parse-community/Parse-SDK-Flutter#custom-objects) (but with extending ParseUser) class and want to get the current user as an instance of your custom class.
> 
> You can use provide a `parseUserConstructor` (only available in [1.0.27](https://github.com/parse-community/Parse-SDK-Flutter/tree/release/1.0.27#custom-objects)) or you can use `ParseUser.currentUser(customUserObject: YourCustomUserClass())`

How can I add this version?

**Comment by maaeps** [comment]
Like this?
`Future<ParseObject> get O async => this.user.objectId == (await ParseUser.currentUser()).objectId ? A : B;`

To use 1.0.27 in your app (not released yet!)
replace
`parse_server_sdk: ^1.0.26`
in your pubspec.yaml with
```
  parse_server_sdk:
    git:
      url: git://github.com/parse-community/Parse-SDK-Flutter.git
      ref: release/1.0.27
```
